inception-ui-kb/src/main/java/de/tudarmstadt/ukp/inception/ui/kb/stmt/editor/StatementEditor.java

```diff
@@ -310,16 +317,31 @@ public void actionQualifierChanged(AjaxQualifierChangedEvent event)
          *            whether the statement being edited is new, meaning it has no corresponding
          *            statement in the KB backend
          */
-        public EditMode(String aId, IModel<KBStatement> aStatement, boolean isNewStatement) {
+         
+        public EditMode(String aId, IModel<KBStatement> aStatement, boolean isNewStatement)
+        {
             super(aId, "editMode", StatementEditor.this, aStatement);
             CompoundPropertyModel<KBStatement> model = CompoundPropertyModel.of(aStatement);
             Form<KBStatement> form = new Form<>("form", model);
             List<ValueType> valueTypes;
-            String rangeValue = property.getObject().getRange();
+            // Set property to the property of the current statement
+            if (property.getObject() == null) {
+                Optional<KBProperty> prop = kbService.readProperty(kbModel.getObject(),
+                    aStatement.getObject().getProperty().getIdentifier());
+                if (prop.isPresent()) {
+                    property.setObject(prop.get());
+                }
+            }
+
+            String rangeValue = null;
+            if (property.getObject() != null) {
+                rangeValue = property.getObject().getRange();
+            }
+
             valueTypes = valueTypeRegistry.getAllTypes();
             if (rangeValue != null) {
-                Optional<KBObject> rangeKBHandle = kbService.
-                        readKBIdentifier(kbModel.getObject().getProject(), rangeValue);
+                Optional<KBObject> rangeKBHandle = kbService
+                        .readKBIdentifier(kbModel.getObject().getProject(), rangeValue);
                 if (rangeKBHandle.isPresent()) {
```

Comment:
Why do you need this? It will create issues if the range is String or INTeger . As We already cover that in every liter support. Any specific reason for this?